### PR TITLE
Fix timezone calculation in Timestamp

### PR DESCRIPTION
**Pull request by cherylEnkidu**

### CODE REVIEW

File: firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/serializers/TimestampSerializer.kt [hunk 1]
```diff
@@ -71,9 +71,7 @@ public object TimestampSerializer : KSerializer<Timestamp> {
     val regex =
       Regex("^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{0,9})?(Z|[+-]\\d{2}:\\d{2})$")
 
-    require(strUppercase.matches(regex)) {
-      "Value does not conform to the RFC3339 specification (str=$str)"
```
Comment (cherylEnkidu) [inline firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/serializers/TimestampSerializer.kt:75]: RFC3339 supports more than 9 digits in time-secfrac, so what is allowed by Firebase is not 100% RFC3339

File: firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/serializers/TimestampSerializer.kt [hunk 2]
```diff
@@ -71,9 +71,7 @@ public object TimestampSerializer : KSerializer<Timestamp> {
...
-    }
+    require(strUppercase.matches(regex)) { "Timestamp string: (str=$str) format doesn't support." }
```
Comment (dconeybe) [inline firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/serializers/TimestampSerializer.kt:74]: This updated error message is less actionable than the old one, as it doesn't give any hints as to what the problem was. If the only digression from RFC3339 is arbitrary sub-second support, then maybe we could just change the old message to something like this:

> Value does not conform to the RFC3339 specification with up to 9 digits of sub-second precision (str=$str)

What do you think?

See go/error-message-best-practices

**Review by dconeybe — CHANGES_REQUESTED** [review]
Thanks for the fix! That's a pretty serious bug. 

Please add a release notes entry to https://github.com/firebase/firebase-android-sdk/blob/eb735a5cd09bb999e9a9444de9a4cf75787ca9d9/firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/FirebaseDataConnect.kt#L83

**Review by dconeybe — APPROVED** [review]

**Comment by dconeybe** [comment]
Thanks for the fix!

**Pull request merged**